### DISCUSSION TITLE
Cleanup old container images

### DISCUSSION THREAD

**Discussion by ajalewis** [siderolabs/talos#9286] [Q&A] [answered]
2 of my nodes currently are running a tad low on disk space. Is there a way I can delete old container images off them? I've had a look in the Docs but I cant seem to find how this is handled (besides resetting the node)?

If I do $ talosctl images list -n $NODE, there are loads there I don't use anymore but there is no remove/delete action?

Thanks

**Comment by smira** [comment, marked as answer]
These images should be automatically cleaned up by the kubelet on disk pressure, please check kubelet logs and settings (if you modified any defaults).

**Reply by samip5** [reply]
But how does one clean it up if kubelet doesn't automatically do that?

I think it should do it...
```
FILESYSTEM   SIZE(GB)   USED(GB)   AVAILABLE(GB)   PERCENT USED   MOUNTED ON
devtmpfs     12.50      0.00       12.50           0.00%          /dev
devshm       12.55      0.00       12.55           0.00%          /dev/shm
efivars      0.00       0.00       0.00            5.40%          /sys/firmware/efi/efivars
overlay      0.00       0.00       0.00            100.00%        /
tmpfs        12.55      0.01       12.54           0.10%          /run
tmpfs        12.55      0.00       12.55           0.00%          /system
tmpfs        0.07       0.00       0.07            0.00%          /tmp
tmpfs        12.55      0.00       12.55           0.00%          /etc/cri/conf.d/hosts
overlay      12.55      0.00       12.55           0.00%          /usr/lib/udev
/dev/sdb6    48.68      35.84      12.84           73.63%         /var
```

**Comment by darkpixel** [comment]
Well this is going to be fun...  One of my nodes is completely out of space due to some bug, and it's so out of space it can't launch the Longhorn storage pods so I can't clean up any volumes.

<img width="711" height="680" alt="Screenshot from 2025-08-23 17-00-35" src="https://github.com/user-attachments/assets/afcd4d5f-1458-48d2-9f6e-40a556a3972e" />
